be/src/olap/rowset/rowset_meta.h

```diff
@@ -0,0 +1,173 @@
+
+#include <memory>
+#include <vector>
+
+#include "olap/new_status.h"
+
+namespace doris {
+
+class RowsetMeta {
+public:
+    virtual void init(const RowsetMetaPb& rowset_meta_pb) {
+        _rowset_meta_pb = rowset_meta_pb;
+    }
+
+    virtual bool deserialize_extra_properties() {
+        return true;
+    }
+
+    virtual void get_rowset_meta_pb(RowsetMetaPb* rowset_meta_pb) {
+        *rowset_meta_pb = _rowset_meta_pb;
+    }
+
+    virtual int64_t get_rowset_id() {
+        return _rowset_meta_pb.rowset_id();
+    }
+
+    virtual void set_rowset_id(int64_t rowset_id) {
+        _rowset_meta_pb.set_rowset_id(rowset_id);
+    }
+
+    virtual int64_t get_tablet_id() {
+        return _rowset_meta_pb.tablet_id();
+    }
+
+    virtual void set_tablet_id(int64_t tablet_id) {
+        _rowset_meta_pb.set_tablet_id(tablet_id);
+    }
+
+    virtual int32_t get_tablet_schema_hash() {
+        return _rowset_meta_pb.tablet_schema_hash();
+    }
+
+    virtual void set_tablet_schema_hash(int64_t tablet_schema_hash) {
+        _rowset_meta_pb.set_tablet_schema_hash(tablet_schema_hash);
+    }
+
+    virtual RowsetType get_rowset_type() {
+        return _rowset_meta_pb.rowset_type();
+    }
+
+    virtual void set_rowset_type(RowsetType rowset_type) {
+        _rowset_meta_pb.set_rowset_type(rowset_type);
+    }
+
+    virtual RowsetState get_rowset_state() {
+        return _rowset_meta_pb.rowset_state();
+    }
+
+    virtual void set_rowset_state(RowsetState rowset_state) {
+        _rowset_meta_pb.set_rowset_state(rowset_state);
+    }
+
+    virtual int get_start_version() {
+        return _rowset_meta_pb.start_version();
+    }
+
+    virtual void set_start_version(int start_version) {
+        _rowset_meta_pb.set_start_version(start_version);
+    }
+    
+    virtual int get_end_version() {
+        return _rowset_meta_pb.end_version();
+    }
+
+    virtual void set_end_version(int end_version) {
+        _rowset_meta_pb.set_end_version(end_version);
+    }
+    
+    virtual int get_row_number() {
+        return _rowset_meta_pb.row_number();
+    }
+
+    virtual void set_row_number(int row_number) {
+        _rowset_meta_pb.set_row_number(row_number);
+    }
+
+    virtual int get_total_disk_size() {
+        return _rowset_meta_pb.total_disk_size();
+    }
+
+    virtual void set_total_disk_size(int total_disk_size) {
+        _rowset_meta_pb.set_total_disk_size(total_disk_size);
+    }
+
+    virtual int get_data_disk_size() {
+        return _rowset_meta_pb.data_disk_size();
+    }
+
+    virtual void set_data_disk_size(int data_disk_size) {
+        _rowset_meta_pb.set_data_disk_size(data_disk_size);
+    }
+
+    virtual int get_index_disk_size() {
+        return _rowset_meta_pb.index_disk_size();
+    }
+
+    virtual void set_index_disk_size(int index_disk_size) {
+        _rowset_meta_pb.set_index_disk_size(index_disk_size);
+    }
+
+    virtual void get_column_statistics(std::vector<ColumnPruning>* column_statistics) {
```

Comment:
On the one side, I don't think ZoneMap is a better name; One the other side, Rename could be done in another cr, because there is many other places using ColumnPruning.

Reply:
ZoneMap is the standard to describe min/max/avg/count statistics.